### PR TITLE
Refactor: IAppArgs, merge CommandLineArgs and ApplicationParametersParser

### PR DESCRIPTION
**Pull request by NickKhalow**
## What does this PR change?

We had 2 objects with a very similar concern, it's unified under the single interface IAppArgs
...

## How to test the changes?

1. Launch the explorer
2. Play Happy path

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md

### CODE REVIEW

**Review by lorux0 — APPROVED** [review]

File: Explorer/Assets/Scripts/Global/Tests/EditMode/RealmLaunchSettingsShould.cs [hunk 1]
```diff
@@ -1,5 +1,6 @@
 using DCL.Browser.DecentralandUrls;
 using DCL.Multiplayer.Connections.DecentralandUrls;
+using Global.AppArgs;
```
Comment (pravusjif) [inline Explorer/Assets/Scripts/Global/Tests/EditMode/RealmLaunchSettingsShould.cs:3]: Is this import needed?

**Review by pravusjif — CHANGES_REQUESTED** [review]
Left a minor request to synch with my recently merged small PR regarding a deep link update, the rest looks good!

File: Explorer/Assets/Scripts/Global/AppArgs/ApplicationParametersParser.cs [hunk 2]
```diff
@@ -45,33 +53,30 @@ private Dictionary<string, string> ParseApplicationParameters(string[] cmdArgs)
...
             }
 
             // in MacOS the deep link string doesn't come in the cmd args...
-#if !UNITY_EDITOR && UNITY_STANDALONE_OSX
-            if (!string.IsNullOrEmpty(Application.absoluteURL) && Application.absoluteURL.StartsWith("decentraland"))
-            {
+#if UNITY_STANDALONE_OSX
+            if (!string.IsNullOrEmpty(Application.absoluteURL ?? string.Empty) && Application.absoluteURL!.StartsWith("decentraland", StringComparison.Ordinal))
+
                 // Regex patch for MacOS removing the ':' from the realm parameter protocol
```
[the remark this answers is not in the document]
Comment (NickKhalow) [reply Explorer/Assets/Scripts/Global/AppArgs/ApplicationParametersParser.cs:63]: Done

File: Explorer/Assets/Scripts/Global/Tests/EditMode/RealmLaunchSettingsShould.cs (hunk 1 above)
Comment (NickKhalow) [reply Explorer/Assets/Scripts/Global/Tests/EditMode/RealmLaunchSettingsShould.cs:3]: Yes

File: Explorer/Assets/Scripts/Global/Dynamic/MainSceneLoader.cs [hunk 3]
```diff
@@ -10,6 +10,7 @@
 using DCL.SceneLoadingScreens.SplashScreen;
 using DCL.Utilities;
 using DCL.Utilities.Extensions;
+using Global.AppArgs;
```
[the remark this answers is not in the document]
Comment (NickKhalow) [reply Explorer/Assets/Scripts/Global/Dynamic/MainSceneLoader.cs:13]: Yes

**Review by pravusjif — APPROVED** [review]

**Pull request merged**